From d689c5e35246dd7ee060023a80fff79bf244e1f9 Mon Sep 17 00:00:00 2001
From: Sascha Weisenberger <sascha.weisenberger@siemens.com>
Date: Mon, 1 Aug 2016 10:35:36 +0100
Subject: [PATCH] Include support for SIMATIC IOT2000 platform

---
 src/x86/x86.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/src/x86/x86.c b/src/x86/x86.c
index b3a0ae5..288a46c 100644
--- a/src/x86/x86.c
+++ b/src/x86/x86.c
@@ -51,7 +51,10 @@ mraa_x86_platform()
     FILE* fh = fopen("/sys/devices/virtual/dmi/id/board_name", "r");
     if (fh != NULL) {
         if (getline(&line, &len, fh) != -1) {
-            if (strncmp(line, "GalileoGen2", 11) == 0) {
+            if (strncmp(line, "SIMATIC IOT2000", 15) == 0) {
+                platform_type = MRAA_INTEL_GALILEO_GEN2;
+                plat = mraa_intel_galileo_gen2();
+            } else if (strncmp(line, "GalileoGen2", 11) == 0) {
                 platform_type = MRAA_INTEL_GALILEO_GEN2;
                 plat = mraa_intel_galileo_gen2();
             } else if (strncmp(line, "BODEGA BAY", 10) == 0) {
-- 
2.1.4

