From 21665847c8b355d9d83c46a5119a7812624ae970 Mon Sep 17 00:00:00 2001
From: Jan Kiszka <jan.kiszka@siemens.com>
Date: Thu, 7 Feb 2019 21:58:05 +0100
Subject: [PATCH 2/3] Switch to x87 mode when targeting ia32

Required for Quark.

Signed-off-by: Jan Kiszka <jan.kiszka@siemens.com>
---
 configure | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/configure b/configure
index ceb04f7998..3da21cf2bf 100755
--- a/configure
+++ b/configure
@@ -1033,6 +1033,9 @@ def configure_v8(o):
   o['variables']['v8_random_seed'] = 0  # Use a random seed for hash tables.
   o['variables']['v8_promise_internal_field_count'] = 1 # Add internal field to promises for async hooks.
   o['variables']['v8_use_snapshot'] = 'false' if options.without_snapshot else 'true'
+  target_arch = options.dest_cpu or host_arch
+  if target_arch == 'ia32':
+    o['variables']['v8_target_arch'] = 'x87'
   o['variables']['v8_trace_maps'] = 1 if options.trace_maps else 0
   o['variables']['node_use_v8_platform'] = b(not options.without_v8_platform)
   o['variables']['node_use_bundled_v8'] = b(not options.without_bundled_v8)
-- 
2.16.4

