diff --git a/tools/Makefile b/tools/Makefile
index a4c4c29..21f2286 100644
--- a/tools/Makefile
+++ b/tools/Makefile
@@ -17,7 +17,7 @@ lib-y += \
 # # Uncomment the next lines to integrate the compiling/linking of
 # # any .c files placed alongside the above "official" tools in the
 # # tools/ directory into SWUpdate's build system.
-# TOOLS_DIR := $(dir $(word $(words $(MAKEFILE_LIST)),$(MAKEFILE_LIST)))
-# TOOLS_SRC := $(wildcard $(TOOLS_DIR)/*.c)
-# TOOLS_BIN := $(patsubst $(TOOLS_DIR)/%.c, %.o, $(TOOLS_SRC))
-# lib-y += $(filter-out $(lib-y),$(TOOLS_BIN))
+TOOLS_DIR := $(dir $(word $(words $(MAKEFILE_LIST)),$(MAKEFILE_LIST)))
+TOOLS_SRC := $(wildcard $(TOOLS_DIR)/*.c)
+TOOLS_BIN := $(patsubst $(TOOLS_DIR)/%.c, %.o, $(TOOLS_SRC))
+lib-y += $(filter-out $(lib-y),$(TOOLS_BIN))
