--- mosquitto-1.4.14/man/manpage.xsl.orig	2018-03-23 10:38:12.765196059 +0800
+++ mosquitto-1.4.14/man/manpage.xsl	2018-03-23 11:10:08.035840449 +0800
@@ -1,6 +1,6 @@
 <!-- Set parameters for manpage xsl -->
 <xsl:stylesheet xmlns:xsl="http://www.w3.org/1999/XSL/Transform" version="1.0">
-	<xsl:import href="/usr/share/xml/docbook/stylesheet/docbook-xsl/manpages/docbook.xsl"/>
+	<xsl:import href="http://docbook.sourceforge.net/release/xsl/current/manpages/docbook.xsl"/>
 	<xsl:strip-space elements="member"/>
 	<!-- Don't display notes list of link urls. -->
 	<xsl:param name="man.endnotes.list.enabled">0</xsl:param>
